govcd/vm.go

```diff
@@ -1508,7 +1530,58 @@ func (vm *VM) UpdateComputePolicy(computePolicy *types.VdcComputePolicy) (*VM, e
 
 }
 
+// UpdateComputePolicyV2Async updates VM Compute policy with the given compute policies using v2.0.0 OpenAPI endpoint,
+// and returns a Task and an error. Updating with an empty compute policy ID will remove it from the VM. Both
```

Comment:
One thing that popped my head and needs clarification - do you know if sending a `types.ComputePolicy{}` structure without placement policy structure (because `types.ComputePolicy.VmPlacementPolicy` field has `omitempty`) still causes removal of placement policy (if it was previously assigned)? 

To work around it, user could at first retrieve ComputePolicy for VM and update only the relevant fields, but we don't yet have vGpu policy fields in `types.ComputePolicy` so it would be impossible to preserve them while doing update. We could possibly add them (without any testing) just to avoid this problem (if my initial query is valid)

vGpu policy fields in `types.ComputePolicy` and that would always wipe it for someone who has manually set it

Reply:
I tested it and sending an empty `&types.ComputePolicy{}` does nothing. I was expecting an error from VCD, because whenever you try to remove all policies from a VM, UI shows an error saying "you need at least one policy". But it simply ignores the call (it performs an update in the VM but the policies remain the same).

Actually this is the reason I added this snippet:

```
	if sizingIsEmpty && placementIsEmpty {
		return Task{}, fmt.Errorf("either sizing policy ID or placement policy ID is needed")
	}
```

As I needed some sort of error to avoid removing all policies in the Terraform provider, otherwise we would get state<->real state mismatches all the time.